homeassistant/components/onvif/config_flow.py

```diff
@@ -225,6 +225,10 @@ async def async_step_profiles(self, user_input=None):
                 device_info = await device_mgmt.GetDeviceInformation()
                 self.device_id = device_info.SerialNumber
 
+            # If no serial number is exposed, fallback to HardwareId
+            if not self.device_id:
+                self.device_id = device_info.HardwareId
```

Comment:
This reads a little strange to me since device info is fetched in the block above, but is also used here. Perhaps remove this block and use an `or` when setting device is above?